proto/outbound.proto

```diff
@@ -308,6 +316,45 @@ message OpaqueRoute {
   }
 }
 
+message TlsRoute {
+  io.linkerd.proxy.meta.Metadata metadata = 1;
+
+  // If empty, the SNI value is ignored.
+  repeated io.linkerd.proxy.tls_route.SniMatch snis = 2;
+
+  // Must have at least one rule.
+  repeated Rule rules = 3;
+  message Rule {
+    Distribution backends = 1;
+  }
+
+  message Distribution {
+    oneof kind {
+      Empty empty = 1;
```

Comment:
Is an empty variant strictly necessary here? We should be able to use presence of the other variants to indicate empty.

Reply:
https://github.com/linkerd/linkerd2-proxy-api/blob/main/proto/outbound.proto#L139 i was curious myself, so here's an existing example of that pattern, just to demonstrate that pattern

Reply:
This is so that this state is explicit and can't be ambiguous i.e. if a field is unset.